### PR TITLE
BLD: GHA migration

### PR DESCRIPTION
**Pull request by tangkong**
## Description
runs pcds-python-migration-tools on this repository 

## Motivation and Context
Prep for the hackathon, clean up this repo 

## How Has This Been Tested?
Hopefully with GHA soon

## Where Has This Been Documented?
This PR

### CODE REVIEW

File: .github/workflows/standard.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  standard:
+    uses: pcdshub/pcds-ci-helpers/.github/workflows/python-standard.yml@master
+    with:
+      # The workflow needs to know the package name.  This can be determined
+      # automatically if the repository name is the same as the import name.
+      package-name: "typhos"
+      # Extras that will be installed for both conda/pip:
+      testing-extras: "happi"
+      # Extras to be installed only for conda-based testing:
+      conda-testing-extras: "pcdsdevices pyqt=5"
```
Comment (klauer) [inline .github/workflows/standard.yml:20]: Let's add pyqt/pyqt5 to the conda-recipe and requirements.txt, respectively
And `pcdsdevices` to the conda recipe

File: conda-recipe/meta.yaml [hunk 2]
```diff
@@ -1,60 +1,74 @@
...
   noarch: python
-  entry_points:
-    - typhos = typhos.cli:main
+  script: {{ PYTHON }} -m pip install . -vv
+
+
 
 requirements:
+  build:
+  - python >=3.7
```
Comment (klauer) [inline conda-recipe/meta.yaml:21]:
```suggestion
  - python >=3.9
```

File: conda-recipe/meta.yaml [hunk 3]
```diff
@@ -1,60 +1,74 @@
...
-  commands:
-    - typhos --help
 
 extra:
   recipe-maintainers:
-    - klauer
-    - hhslepicka
-    - zllentz
+  - klauer
+  - hhslepicka
```
Comment (klauer) [inline conda-recipe/meta.yaml:73]:
```suggestion
  - tangkong
```

(Or just remove the extra section entirely; it's a conda-forge thing)

File: typhos/utils.py [hunk 4]
```diff
@@ -248,10 +247,10 @@ def use_stylesheet(dark=False, widget=None):
         # Load the path to the file
         style_path = os.path.join(ui_dir, 'style.qss')
         if not os.path.exists(style_path):
-            raise EnvironmentError("Unable to find Typhos stylesheet in {}"
```
Comment (klauer) [inline typhos/utils.py:251]: First I've noticed it:
```
The following exceptions are kept for compatibility with previous versions; starting from Python 3.3, they are aliases of [OSError](https://docs.python.org/3/library/exceptions.html#OSError).

exception EnvironmentError
```
https://docs.python.org/3/library/exceptions.html#EnvironmentError

File: conda-recipe/meta.yaml (hunk 3 above)
Comment (tangkong) [reply conda-recipe/meta.yaml:86]: I'll enjoy having my name there :)

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,60 @@
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = [ "setuptools>=45", "setuptools_scm[toml]>=6.2",]
+
+[project]
+classifiers = [ "Development Status :: 2 - Pre-Alpha", "Natural Language :: English", "Programming Language :: Python :: 3",]
```
Comment (ZLLentz) [inline pyproject.toml:6]: Typhos shouldn't be pre-alpha I think

File: requirements.txt [hunk 6]
```diff
@@ -4,6 +4,7 @@ numpy
 numpydoc
 ophyd
 pcdsutils
+PyQt5
```
Comment (klauer) [inline requirements.txt:7]: `dev_requirements` was unchanged here, commenting on this file as it's closest. Before I forget:

`dev_requirements.txt`:
* `doctr` needs to be removed
* `doctr-verisons-menu` needs to be removed

`docs_requirements.txt` needs to be added with:
* sphinx and the other docs-building requirements
* `docs-versions-menu`
*

File: pyproject.toml (hunk 5 above)
Comment (tangkong) [reply pyproject.toml:6]: Probably [Development Status :: 5 - Production/Stable](https://pypi.org/search/?c=Development+Status+%3A%3A+5+-+Production%2FStable)?

File: docs-requirements.txt [hunk 7]
```diff
@@ -0,0 +1,3 @@
+docs-versions-menu
```
Comment (klauer) [inline docs-requirements.txt:1]: Move `sphinxcontrib-jquery` over too

Comment (tangkong) [reply docs-requirements.txt:1]: Thank you, lunch brain

File: pyproject.toml (hunk 5 above)
Comment (ZLLentz) [reply pyproject.toml:6]: I think so, I should have googled the options myself

Comment (tangkong) [reply pyproject.toml:6]: No worries.  That hyperlink does not go where I thought it did

File: typhos/tests/test_console.py [hunk 8]
```diff
@@ -17,59 +17,59 @@ def test_base_console(qtbot):
     tc.shutdown()
 
 
-def test_add_happi_device(qapp, qtbot, happi_cfg, client):
-    device = client['Syn:Motor'].get()
+# def test_add_happi_device(qapp, qtbot, happi_cfg, client):
```
Comment (klauer) [inline typhos/tests/test_console.py:20]: Maybe
```
# top-level module:
import pytest

pytest.skip()
```

Comment (tangkong) [reply typhos/tests/test_console.py:20]: I was hoping that I might find the solution to these test suite problems, but if I give up I'll `pytest.skip()`

**Comment by tangkong** [comment]
I give up on the tests for now.  I think for the sake of allowing people to contribute to typhos (and not having to deal with pyupgrade merge conflicts) I'll make a followup issue for the troublesome tests.  Maybe I'll get local gha working some time soon.

**Comment by ZLLentz** [comment]
I think that's a good call.
To be clear, this issue only happens on the CI, not when running the tests locally?

**Comment by tangkong** [comment]
> I think that's a good call. To be clear, this issue only happens on the CI, not when running the tests locally?

Yep, all the tests pass for me locally.  😢

**Comment by ZLLentz** [comment]
I wonder if there's any way to tell if we're on the CI so we can at least make sure they all run locally. A generic @skip_on_ci decorator could be useful.

**Review by ZLLentz — APPROVED** [review]
Do we need to merge before we can set the required/not required settings?

**Comment by klauer** [comment]
No, that can be set right now: https://github.com/pcdshub/pcds-python-migration-tools/blob/master/update_github_settings.py

**Comment by tangkong** [comment]
huh we didn't have branch protection rules already set up for typhos?...

**Comment by klauer** [comment]
Branch protection set with the above-mentioned script.

Merging because our required tests are passing and it's Friday and there's no one here to stop me.

**Pull request merged**